processing/src/main/java/io/druid/query/MetricsEmittingExecutorService.java

```diff
@@ -23,14 +23,49 @@
 import com.metamx.emitter.service.ServiceEmitter;
 import com.metamx.emitter.service.ServiceMetricEvent;
 
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
 import java.util.concurrent.Callable;
 
 public class MetricsEmittingExecutorService extends ForwardingListeningExecutorService
 {
+  private static final ThreadMXBean THREAD_MX_BEAN = ManagementFactory.getThreadMXBean();
+  private static final ThreadLocal<Long> THREAD_USER_START = new ThreadLocal<Long>()
+  {
+    @Override
+    public Long initialValue()
+    {
+      return 0L;
+    }
+  };
+  private static final Runnable PRE = new Runnable()
+  {
+    @Override
+    public void run()
+    {
+      if (THREAD_MX_BEAN.isCurrentThreadCpuTimeSupported()) {
```

Comment:
instead of doing this if here, I think it would be more useful to initialize these runnables in a `static {}` initialization block that checks for support and, if supported, attaches `Runnable` objects that use it without question.  If not supported, it can build noop `Runnable` objects.

This is a micro-optimization, but, in general, any time you are asking a question of your system to figure out behavior, it's nicest to ask the question once and build objects that line up with the correct assumptions.  JIT can then do code-level optimization.

In this case, we would likely get the same performance gains out of branch prediction, but where it is easy, making the choice once and then just operating according to that choice should do better things.